waku/v2/node/wakunode2.go

```diff
@@ -751,10 +751,7 @@ func (w *WakuNode) DialPeerWithInfo(ctx context.Context, peerInfo peer.AddrInfo)
 
 func (w *WakuNode) connect(ctx context.Context, info peer.AddrInfo) error {
 	err := w.host.Connect(ctx, info)
-	if err != nil {
-		w.host.Peerstore().(wps.WakuPeerstore).AddConnFailure(info.ID)
-		return err
-	}
+	w.peermanager.HandleDialError(err, info.ID)
```

Comment:
maybe add a check for peermanager nil, afair PeerManager is optional to start WakuNode and can lead to panic.

Reply:
Fixed in last commit